packages/gatsby/src/utils/worker/render-html.js

```diff
@@ -25,7 +15,12 @@ export function renderHTML({ htmlComponentRendererPath, paths, envVars }) {
         const htmlComponentRenderer = require(htmlComponentRendererPath)
         try {
           htmlComponentRenderer.default(path, (throwAway, htmlString) => {
-            resolve(fs.outputFile(generatePathToOutput(path), htmlString))
+            resolve(
+              fs.outputFile(
+                getPageHtmlFilePath(join(process.cwd(), `public`), path),
+                htmlString
+              )
```

Comment:
just interested:

everywhere there is `publicDir` as param, but here is ```process.cwd(), `public`)``` used

i think the publicDir comes from here?
https://github.com/interactive-investor/gatsby/blob/0be16c63cc4da72f9d1049c8aa6d737926d62dfd/packages/gatsby/src/commands/build.js#L49-L50

Reply:
Where `renderHTML` used in `commands/build-html.ts` `doBuildPages` program is not passed the parameter.   

https://github.com/gatsbyjs/gatsby/blob/7527993bb575563d54918b7ea90939d7bc9e4975/packages/gatsby/src/commands/build-html.ts#L137

maybe this is to avoid `program` being passed unnecessarily to other functions.

Reply:
We actually should be passing `program` around (or at least build directory) instead of using `process.cwd()`, but this doesn't change anything in practice for now (as `process.cwd()` was already used before).

We do want to make it so we don't use `process.cwd()` at all, as this make it hard for us to enable options like changing build directory in future, or even things like triggering builds programmatically (instead of using CLI), but this is not concern of this PR, so it's ok for it to stay like that for now

Reply:
I have noted this in mind of future change of output dir

Yes - it is not the scope of this PR